src/SchemaDesign/SchemaDesigner.tsx

```diff
@@ -0,0 +1,416 @@
+import { Schema, SchemaQuestion, SplitSchema, SchemaType, Step } from '@/types/interfaces'
+import { withTheme } from '@rjsf/core'
+import Box from '@mui/material/Box'
+import Button from '@mui/material/Button'
+import Divider from '@mui/material/Divider'
+import Paper from '@mui/material/Paper'
+import Stack from '@mui/material/Stack'
+import Tab from '@mui/material/Tab'
+import Tabs from '@mui/material/Tabs'
+import TextField from '@mui/material/TextField'
+import Typography from '@mui/material/Typography'
+import React, { useEffect, useState } from 'react'
+import { DragDropContext, Droppable, Draggable, DraggableProvided } from 'react-beautiful-dnd'
+import { getStepsFromSchema } from '@/utils/formUtils'
+import InputLabel from '@mui/material/InputLabel'
+import MenuItem from '@mui/material/MenuItem'
+import Select from '@mui/material/Select'
+import Icon from '@mui/material/Icon'
+import AddIcon from '@mui/icons-material/Add'
+import TextFieldsIcon from '@mui/icons-material/TextFields'
+import CheckBoxIcon from '@mui/icons-material/CheckBox'
+import CalendarMonthIcon from '@mui/icons-material/CalendarMonth'
+import DialogActions from '@mui/material/DialogActions'
+import DialogContent from '@mui/material/DialogContent'
+import Dialog from '@mui/material/Dialog'
+import DialogTitle from '@mui/material/DialogTitle'
+import DialogContentText from '@mui/material/DialogContentText'
+import _ from 'lodash'
+import Grid from '@mui/material/Grid'
+import Alert from '@mui/material/Alert'
+import IconButton from '@mui/material/IconButton'
+import DeleteIcon from '@mui/icons-material/Clear'
+import { Theme as MaterialUITheme } from '../MuiForms'
+import QuestionPicker from './QuestionPicker'
+
+const SchemaForm = withTheme(MaterialUITheme)
+
+export default function SchemaDesigner() {
+  const [questionPickerOpen, setQuestionPickerOpen] = useState(false)
+  const [steps, setSteps] = useState<any[]>([])
```

Comment:
I seem to remember `steps` being awkward to type. Is it worth us creating a `Step` type that simply extends `any`, but comes with an internal comment explaining why we have to type it as such? That way we could at least be consistent with using the `Step` type and reduce the number of `any`s we use. I know it makes littler difference functionally, but at least a new developer could understand why we use `any` and what the value is meant to represent by reading the comment.

As this would be a change that affects code out of the scope of this ticket I'd suggest making this change in a separate ticket, if you agree with my reasoning of course.

Reply:
I think that you're right that this should be moved to a new ticket. Created `BAI-388` to look into this!